packages/build/bin/compile-package.js

```diff
@@ -128,7 +133,76 @@ function run(argv, options) {
 
   args.push(...compilerOpts);
 
-  return utils.runCLI('typescript/lib/tsc', args, {cwd, ...options});
+  const validArgs = validArgsForBuild(args);
+
+  return utils.runCLI('typescript/lib/tsc', validArgs, {cwd, ...options});
+}
+
+/**
+ * `tsc -b` only accepts valid arguments. `npm run build -- --<other-arg>` may
+ * pass in extra arguments. We need to remove such arguments.
+ * @param {string[]} args An array of arguments
+ */
+function validArgsForBuild(args) {
+  const validBooleanOptions = [
+    '-h',
+    '--help',
+    '-w',
+    '--watch',
+    '--preserveWatchOutput',
+    '--listFiles',
+    '--listEmittedFiles',
+    '--pretty',
+    '--traceResolution',
+    '--diagnostics',
+    '--extendedDiagnostics',
+    '-i',
+    '--incremental',
+    '--assumeChangesOnlyAffectDirectDependencies',
+    '-v',
+    '--verbose',
+    '-d',
+    '--dry',
+    '-f',
+    '--force',
+    '--clean',
+  ];
```

Comment:
Is there a way how to get this list from `tsc`, so that we don't have to maintain it manually?

Reply:
No, I digged into `tsc.js`. The list of options is not exported. :-(